src/gym_electric_motor/physical_systems/electric_motors/six_phase_motor.py

```diff
@@ -0,0 +1,99 @@
+    _t46= 1/ 3 * np.array([
+        [1, -0.5, -0.5, 0.5 * np.sqrt(3), -0.5 * np.sqrt(3), 0],
+        [0, 0.5 * np.sqrt(3), -0.5 * np.sqrt(3), 0.5, 0.5, -1],
+        [1, -0.5, -0.5, -0.5 * np.sqrt(3), 0.5 * np.sqrt(3), 0],
+        [0, -0.5 * np.sqrt(3), 0.5 * np.sqrt(3), 0.5, 0.5, -1]
+    ])
+
+    # i_s - stator current
+    @staticmethod
+    def t_46(quantities):
+        """
+        Transformation from abc representation to alpha-beta representation
+
+        Args:
+            quantities: The properties in the abc representation like ''[i_sa1, i_sb1, i_sc1, i_sa2, i_sb2, i_sc2]''
+
+        Returns:
+            The converted quantities in the alpha-beta representation like ''[i_salpha, i_sbeta, i_sX, i_sY]''
+        """
+        return np.matmul(SixPhaseMotor._t46, quantities)
+
+    @staticmethod
+    def q(quantities, epsilon):
+        """
+        Transformation of the abc representation into dq using the electrical angle
+
+        Args:
+            quantities: the properties in the abc representation like ''[i_sa1, i_sb1, i_sc1, i_sa2, i_sb2, i_sc2]''
+            epsilon: electrical rotor position
+
+        Returns:
+            The converted quantities in the dq representation like ''[i_sd, i_sq, i_sx, i_sy, i_s0+, i_s0-]''.
+            since 2N topology is considered (case where the neutral points are not connected) i_s0+, i_s0- will not be taken into account 
+        """
+        """
+        t_vsd = 1/ 3 * np.array([
+            [1, -0.5, -0.5, 0.5 * np.sqrt(3), -0.5 * np.sqrt(3), 0],
+            [0, 0.5 * np.sqrt(3), -0.5 * np.sqrt(3), 0.5, 0.5, -1],
+            [1, -0.5, -0.5, -0.5 * np.sqrt(3), 0.5 * np.sqrt(3), 0],
+            [0, -0.5 * np.sqrt(3), 0.5 * np.sqrt(3), 0.5, 0.5, -1],
+            [1, 1, 1, 0, 0, 0],
+            [0, 0, 0, 1, 1, 1]
+        ])
+        tp_alphaBetaXY = np.array([
+                 [cos, sin, 0, 0, 0, 0],
+                 [-sin, cos, 0, 0, 0, 0],
+                 [0, 0, cos, sin, 0, 0],
+                 [0, 0, -sin, cos, 0, 0],
+                 [0, 0, 0, 0, 1, 0]
+                 [0, 0, 0, 0, 0, 1]
+        ])
+        """
+        t_vsd = 1/ 3 * np.array([
+            [1, -0.5, -0.5, 0.5 * np.sqrt(3), -0.5 * np.sqrt(3), 0],
+            [0, 0.5 * np.sqrt(3), -0.5 * np.sqrt(3), 0.5, 0.5, -1],
+            [1, -0.5, -0.5, -0.5 * np.sqrt(3), 0.5 * np.sqrt(3), 0],
+            [0, -0.5 * np.sqrt(3), 0.5 * np.sqrt(3), 0.5, 0.5, -1],
+        ])
+
+        def rotation_matrix(theta):
+           return np.array([
+        [math.cos(theta), math.sin(theta)],
+        [-math.sin(theta), math.cos(theta)]
+        ])
+        
+        t1 = rotation_matrix(epsilon)
+        t2 = rotation_matrix(-epsilon)
+        tp_alphaBetaXY = np.block([
+            [t1, np.zeros(t1.shape[0],t1.shape[1])],
+            [np.zeros(t2.shape[0],t2.shape[1]), t2],
+        ])
```

Comment:
scipy provides a command that should make this snippet more clear

https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.block_diag.html